File: top-interview-150/26_Remove_duplicates_from_sorted_array.cs
// 26. Remove Duplicates from Sorted Array

// Given an integer array nums sorted in non-decreasing order, remove the duplicates in-place such that each unique element appears only once. The relative order of the elements should be kept the same. Then return the number of unique elements in nums.
// Consider the number of unique elements of nums to be k, to get accepted, you need to do the following things:
    // Change the array nums such that the first k elements of nums contain the unique elements in the order they were present in nums initially. The remaining elements of nums are not important as well as the size of nums.
    // Return k.

public class Solution {
    public int RemoveDuplicates(int[] nums) {
        int i = 0;
        int j = 1;
        int k = 0;
        int limit = nums.Length - 1;

        if(limit >= 0){
            k = 1;
            while(j <= limit){
                if(nums[i] == nums[j]){
                    j++;
                }else{
                    i++;
                    nums[i] = nums[j];
                    k++;
                }
            }
        }

        return k;

    }
}

File: top-interview-150/Array-String/26_Remove_duplicates_from_sorted_array.cpp
// 26. Remove Duplicates from Sorted Array

// Given an integer array nums sorted in non-decreasing order, remove the duplicates in-place such that each unique element appears only once. The relative order of the elements should be kept the same.
// Consider the number of unique elements in nums to be k​​​​​​​​​​​​​​. After removing duplicates, return the number of unique elements k.
// The first k elements of nums should contain the unique numbers in sorted order. The remaining elements beyond index k - 1 can be ignored.

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i = 1;
        int last = 0;

        if(nums.size() == 1){
            return 1;
        }

        while(i < nums.size()){
            if(nums[i] == nums[last]){
                i++;
            }else{
                last++;
                nums[last] = nums[i];
            }
        }
        last++;

        return last;
    }
};